proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -668,4 +667,33 @@ public boolean isConnected()
     {
         return !ch.isClosed();
     }
+
+    public void relayMessage(PluginMessage input) throws Exception
+    {
+        if ( input.getData().length >= Byte.MAX_VALUE )
+        {
+            return;
+        }
+
+        if ( input.getTag().equals( "REGISTER" ) || input.getTag().equals( "minecraft:register" ) )
+        {
+            String content = new String( input.getData(), StandardCharsets.UTF_8 );
+
+            for ( String id : content.split( "\0" ) )
+            {
+                relayRegisteredChannels.add( id );
```

Comment:
Need to limit ID length, especially if the other limit is going to be removed.

Reply:
What about 128 characters max and 128 registered channels max ? That's 16k bytes max per user

Reply:
That other limit shouldn't be removed, just edited to 16k. The limits on channel count and channel name length seem good, but maybe we could have a general limit of 16k but allow 256 channels and a channel name length of 256 as that doesn't hurt us